### PR TITLE
Add backprop for Softmax

### PR DESCRIPTION
**Pull request by vuoristo**
Implements backprop for Softmax graph op.

---
This change is [Reviewable](https://reviewable.io/reviews/pair-code/deeplearnjs/427)

### CODE REVIEW

**Comment by vuoristo** [comment]
Updated the test to run with 0.3.15.

Btw, the backprop logic is from here https://github.com/tensorflow/tensorflow/blob/97a4c226e8a9e7c5c36fc38e2b9f8459c77abd5a/tensorflow/python/ops/nn_grad.py#L206

**Comment by nsthorat** [comment]
:lgtm_strong:

Nice! Thanks for the contribution :)

---

Review status: 0 of 2 files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/pair-code/deeplearnjs/427#-:-L1TAlmQpyVns8p_knFR:b-q2n105)*
